webapp/src/actions.js

```diff
@@ -139,3 +143,24 @@ export const enqueue = (id) => async (dispatch, getState) => {
     dispatch(list(false, 'my'));
     dispatch(list(false, 'in'));
 };
+
+export const bump = (id) => async (dispatch, getState) => {
+    await fetch(getPluginServerRoute(getState()) + '/bump', {
+        method: 'POST',
+        credentials: 'same-origin',
+        headers: {
+            'Content-Type': 'application/json',
+            'X-Requested-With': 'XMLHttpRequest',
+        },
```

Comment:
I think you get the same-origin and headers for free if import client4 from mattermost-redux (see then client.js file from workflow).

Reply:
Regarding Typescript, I think it would be a bit out of scope, but I can try. Nevertheless, I feel makes more sense a new PR to migrate the whole plugin. Also, the code is mostly from webapp, but I removed some things that felt unneeded for this implementation, or that included too many more components from webapp.

Regarding the headers, solved in the new commit.

Reply:
Yep, don't worry about Typescript for now. It's for another PR.